libs/template/templates/dbt-sql/template/{{.project_name}}/profile_template.yml.tmpl

```diff
@@ -0,0 +1,32 @@
+# This file defines prompts with defaults for dbt initializaton.
+# It is used when the `dbt init` command is invoked.
+#
+fixed:
+  type: databricks
+prompts:
+  host:
+    default: {{(regexp "^https?://").ReplaceAllString .workspace_host_override ""}}
+  token:
+    hint: 'personal access token to use, dapiXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
+    hide_input: true
+  {{- if eq .catalog ""}}
+  _choose_unity_catalog:
+    'use Unity Catalog':
+      catalog:
+        hint: 'initial catalog'
+    'do not use Unity Catalog':
+      _fixed_catalog: null
+  {{- else}}
+  http_path:
+    default: {{.http_path}}
+  catalog:
+    hint: 'initial catalog'
+    default: {{.catalog}}
+  {{- end}}
+  schema:
+    hint: 'default schema where dbt will build objects, e.g. lennart_dev'
+    default: {{.schema}}
```

Comment:
How could we parameterize this for different targets? I.e. can this be passed in through a flag or env var and then specified in the job configuration?

Reply:
The schema value that you see here is only the "default" for local development. This default is used as follows:

1. Customers use `databricks bundle init` to create a new project using the dbt-sql template. They select a default catalog / schema there.
2. To setup their personal, local development environment, customers use `dbt init` (following the instructions in the README). dbt will ask for the workspace host (which defaults to `{{workspace_host}}`), the catalog (which defaults to `{{.catalog}}`, the schema (which defaults to {{.schema}}`). Now, some customers might use the same schema for all of dev. Others may use a personal schema like lennart_dev (which is common in the dbt community).

Note that an alternative to `dbt init`, some savvy customers might just directly edit their ~/.dbt/profiles.yml.

Reply:
Discussed offline: we should add some messaging to this template: likely a comment that links to a GitHub issue that tracks our support for a setting the catalog/schema.

Reply:
So I did another major pass over this, now offering an enum option so users can decide to use a personal schema such as `pieter_noordhuis`. PTAL.